### PR TITLE
Sort mentor inbox

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Note: this PR builds on https://github.com/exercism/v3-website/pull/63, so only merged this  PR _after_ merging #63

### CODE REVIEW

File: app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def index
         trackTitle: "Ruby",
         trackIconUrl: "https://assets.exercism.io/tracks/ruby-hex-white.png",
         menteeAvatarUrl: "https://robohash.org/exercism",
-        menteeHandle: "mentee",
+        menteeHandle: "Mentee",
```
Comment (ErikSchierboom) [inline app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb:11]: I've done this just to have the sorting make sense without having to do case-insensitive sorting :D

File: app/javascript/components/mentoring/mentor_inbox.jsx [hunk 2]
```diff
@@ -28,9 +34,14 @@ export function MentorInbox({ tracksRequest, ...props }) {
...
   }
 
+  function setSort(sort) {
+    dispatch({ type: 'sort.changed', payload: { sort: sort } })
+  }
+
   return (
     <div>
       <TrackFilter request={tracksRequest} setTrack={setTrack} />
+      <Sorting sort={conversationsRequest.query.sort} setSort={setSort} />
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/mentor_inbox.jsx]: I'm not sure about the naming of this component. Let me know what you think.

File: app/javascript/components/mentoring/mentor_inbox/sorting.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+const sortingOptions = [
+  { value: 'exerciseTitle', description: 'Sort by Exercise' },
+  { value: 'menteeHandle', description: 'Sort by Student' },
+]
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/mentor_inbox/sorting.jsx]: I've hardcoded the sorting options here. We could pass them as parameters, although I probably wouldn't want to pass their description too as that is a UI responsibility.

File: app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+
+const sortingOptions = [
```
Comment (kntsoriano) [inline app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx:3]: Might be for another PR, but should we send this via the API as well? It allows us to introduce more sorting options server side without worrying about our JS.

File: app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb (hunk 1 above)
Comment (kntsoriano) [reply app/controllers/test/components/mentoring/mentor_inboxes/conversations_controller.rb:11]: Ha. Yes!

**Review by kntsoriano — APPROVED** [review]
Looks great!

File: app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx (hunk 4 above)
Comment (ErikSchierboom) [reply app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx:3]: Yeah that would be a good idea for a follow-up PR. Would you include both the value and the description in the API?

Comment (kntsoriano) [reply app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx:3]: Yes. I think we should include both so that the client side code is really flexible.

Comment (ErikSchierboom) [reply app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx:3]: Woudl you include it in the existing API call that retrieves the conversation list or create a new API endpoint?

Comment (kntsoriano) [reply app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx:3]: Oh man. Good question. I don't know. Should we go with whats easier right now?

Comment (ErikSchierboom) [reply app/javascript/components/mentoring/mentor_inbox/conversation_sorter.jsx:3]: I'm fine with either option :)
